cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/types/TypeParser.java

```diff
@@ -598,7 +554,7 @@ private static List<String> joinPrimitive(@NotNull List<String> typeBlocks) {
     boolean foundPrimitive = false;
 
     for (String s : typeBlocks) {
-      if (PRIMITIVES.contains(s)) {
+      if (getLanguage().getPrimitiveTypes().contains(s)) {
```

Comment:
Can we add language as an additional parameter here? I want to get rid of the `getLanguage` calls in the singleton type manager.